### PR TITLE
Add mariadb backup as optional option for wsrep_sst_method

### PR DESCRIPTION
**Pull request by eRadical**
mariabackup is an alternative to rsync that does not block the donor node

### CODE REVIEW

**Comment by mrlesmithjr** [comment]
@eRadical - I merged some PR's from @elcomtik which caused some conflicts with this PR. Do you mind rebasing to resolve these and we can get this PR merged as well?

**Comment by elcomtik** [comment]
@eRadical Did it worked? I use mariabackup when SST encryption is needed. What I found is that mariabackup needs to authenticate on the donor node, so I had to create a user for that. I don't see that you use any SST authentication.

You may see respective code at https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/1e263fa31c2a4163d3cd6d13f46f210801b926f8/defaults/main.yml#L159-L165
and https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/1e263fa31c2a4163d3cd6d13f46f210801b926f8/tasks/mysql_users.yml#L20 and in any galera config file, eg. https://github.com/mrlesmithjr/ansible-mariadb-galera-cluster/blob/1e263fa31c2a4163d3cd6d13f46f210801b926f8/templates/etc/my.cnf.d/server.cnf.j2#L85

**Comment by eRadical** [comment]
Haven't had any time yesterday as i was distracted by other chores.
Today I'll spin up a cluster w/ the above to test.

Thanx @elcomtik for those. Will incorporate!

**Comment by eRadical** [comment]
You're right @elcomtik, it does need authentication.

I've changed the logic to create user & use mariabackup without tls.
The tls part is still there but can be enabled separately.

What do you think about this form?

**Comment by elcomtik** [comment]
This looks almost perfect, although it removed functionality to automatically use mariabackup if sst encryption is enabled.
I would like if variable galera_sst_method default would be conditional which will switch on mariabackup if SST TLS is used.

Something like this
```
galera_sst_method = "mariadb_tls_files and mariadb_tls_files|length == 3 and galera_sst_tls_enabled | ternary( "mariabackup", "rsync" )"
```

This could be overridden by the user if needed

**Comment by eRadical** [comment]
In defaults/main.yml I would change `galera_sst_method` to `galera_sst_default_method`:
```
galera_sst_default_method: rsync
```

In vars/main.yaml I would add:
```
galera_sst_method: "{{ mariadb_tls_files and mariadb_tls_files|length == 3 and galera_sst_tls_enabled | ternary( "mariabackup", galera_sst_default_method ) }}"
```

Thoughts ? cc @mrlesmithjr

**Comment by eRadical** [comment]
Done as @elcomtik suggested!

@mrlesmithjr - this is ready to be merged!

I have some clusters to spin up! :)

File: defaults/main.yml [hunk 1]
```diff
@@ -164,6 +163,13 @@ mariadb_sst_user:
     password: "{{ mariadb_sst_password }}"
     priv: "*.*:RELOAD,PROCESS,LOCK TABLES,BINLOG MONITOR"
 
+# This option defines the wsrep_sst_method that should be used by the cluster.
+# Possible options:
+#  - mariabackup - recommended for MariaDB - default if TLS is enabled via galera_sst_tls_enabled,
+#  - xtrabackup-v2 - there are limitations, please see https://mariadb.com/kb/en/xtrabackup-v2-sst-method/
```
Comment (elcomtik) [inline defaults/main.yml:169]: we should not list xtrabackup-v2, because install of respective packages is not handled yet

**Review by elcomtik — COMMENTED** [review]
Looks, good. Please look at my last comment about xtrabackup-v2

File: defaults/main.yml (hunk 1 above)
Comment (gabriel-preda-adswizz) [reply defaults/main.yml:169]: Or we could just say it's unsupported by the role!

Comment (eRadical) [reply defaults/main.yml:169]: Or we could just say it's unsupported by the role!

**Comment by elcomtik** [comment]
@eRadical I just found an issue with WSREP auth. It affects your PR, you should rebase it when my PR is merged.

**Comment by eRadical** [comment]
As soon as @mrlesmithjr can help us merge both of them!

**Comment by mrlesmithjr** [comment]
Merged #68

**Comment by eRadical** [comment]
rebased and added also a note that xtrabackup-v2 is not implemented by this role

**Comment by elcomtik** [comment]
Perfect job @eRadical. LGTM @mrlesmithjr

**Comment by mrlesmithjr** [comment]
Merged and thanks to you both as always!

**Pull request merged**